src/deploy/scope_resourcegroup.ts

```diff
@@ -15,6 +15,11 @@ export async function DeployResourceGroupScope(azPath: string, validationOnly: b
         throw Error(`Resource Group ${resourceGroupName} could not be found.`)
     }
 
+    if (deploymentMode === "validate") {
+        // updating mode as validate is not a valid mode value for Az cli command
+        deploymentMode = "Incremental"
```

Comment:
this is problematic as you will never hit the condition in L64 and L70.

you should just set the `azDeployParameters` appropriately

```
    const azDeployParameters = [
        resourceGroupName ? `--resource-group ${resourceGroupName}` : undefined,	        resourceGroupName ? `--resource-group ${resourceGroupName}` : undefined,
        template ?	        template ?
            template.startsWith("http") ? `--template-uri ${template}` : `--template-file ${template}`	            template.startsWith("http") ? `--template-uri ${template}` : `--template-file ${template}`
            : undefined,	            : undefined,
        deploymentMode && deploymentMode != "validate" ? `--mode ${deploymentMode}` : "--mode Incremental",	        deploymentMode ? `--mode ${deploymentMode}` : "Incremental",
        deploymentName ? `--name "${deploymentName}"` : undefined,	        deploymentName ? `--name "${deploymentName}"` : undefined,
        parameters ? `--parameters ${parameters}` : undefined	        parameters ? `--parameters ${parameters}` : undefined
    ].filter(Boolean).join(' ');	    ].filter(Boolean).join(' ');
```

Reply:
BTW there is an error in line L29 which I have fixed above. In else case, we were not adding `--mode`